File: backend/services/layout_service.py
"""
Layout Service
Handles business logic for menu and header nav; uses layout_repository.
"""
from repositories.layout_repository import get_menu as repo_get_menu, get_header_nav as repo_get_header_nav


def get_menu():
    """Return sidebar menu items."""
    return repo_get_menu()


def get_header_nav():
    """Return header navigation items."""
    return repo_get_header_nav()

File: frontend/app/services/layoutService.ts
/**
 * Service for layout data (header nav, etc.).
 */
import { apiGet } from './api';

export async function getHeaderNav(): Promise<Array<{ id: string; label: string; href: string }>> {
  const data = (await apiGet('/api/layout/header-nav')) as { items?: Array<{ id: string; label: string; href: string }> };
  return data.items ?? [];
}
